File: RuyiSDKCpp/RuyiNet/Service/Leaderboard/RuyiNetLeaderboardInfo.cpp
#include "RuyiNetLeaderboardInfo.h"

namespace Ruyi { namespace SDK { namespace Online {

	RuyiNetLeaderboardInfo::RuyiNetLeaderboardInfo() {}
	RuyiNetLeaderboardInfo::RuyiNetLeaderboardInfo(RuyiNetGetGlobalLeaderboardVersionsResponse::Data& data)
	{
		LeaderboardId = data.leaderboardId;
		LeaderboardType = ConvertStringToRuyiNetLeaderboardType(data.leaderboardType);
		RotationType = ConvertStringToRuyiNetRotationType(data.rotationType);
		RetainedCount = data.retainedCount;

		std::for_each(data.versions.begin(), data.versions.end(), [&](RuyiNetGetGlobalLeaderboardVersionsResponse::Data::VersionInfo& info)
		{
			Versions.push_back(new RuyiNetLeaderboardVersionInfo(info));
		});
	}

	void RuyiNetLeaderboardInfo::GetData(RuyiNetGetGlobalLeaderboardVersionsResponse::Data& data)
	{
		LeaderboardId = data.leaderboardId;
		LeaderboardType = ConvertStringToRuyiNetLeaderboardType(data.leaderboardType);
		RotationType = ConvertStringToRuyiNetRotationType(data.rotationType);
		RetainedCount = data.retainedCount;

		std::for_each(data.versions.begin(), data.versions.end(), [&](RuyiNetGetGlobalLeaderboardVersionsResponse::Data::VersionInfo& info)
		{
			Versions.push_back(new RuyiNetLeaderboardVersionInfo(info));
		});
	}

	std::string RuyiNetLeaderboardInfo::GetLeaderboardId() { return LeaderboardId; }
	RuyiNetLeaderboardType RuyiNetLeaderboardInfo::GetLeaderboardType() { return LeaderboardType; }
	RuyiNetRotationType RuyiNetLeaderboardInfo::GetRotationType() { return RotationType; }
	int RuyiNetLeaderboardInfo::GetRetainedCount() { return RetainedCount; }
	std::vector<RuyiNetLeaderboardVersionInfo*>& RuyiNetLeaderboardInfo::GetVersions() { return Versions; }


}}}

File: RuyiSDKAsync/RuyiNet/Service/Leaderboard/RuyiNetLeaderboardInfo.cs
using System;
using System.Collections.Generic;

namespace Ruyi.SDK.Online
{
    /// <summary>
    /// Represents a leaderboard on RuyiNet.
    /// </summary>
    public class RuyiNetLeaderboardInfo
    {
        /// <summary>
        /// Construct from response data.
        /// </summary>
        /// <param name="data">The data returned from the response.</param>
        public RuyiNetLeaderboardInfo(RuyiNetGetGlobalLeaderboardVersionsResponse.Data data)
        {
            LeaderboardId = data.leaderboardId;
            LeaderboardType = (RuyiNetLeaderboardType)Enum.Parse(typeof(RuyiNetLeaderboardType), data.leaderboardType);
            RotationType = (RuyiNetRotationType)Enum.Parse(typeof(RuyiNetRotationType), data.rotationType);
            RetainedCount = data.retainedCount;

            Versions = new List<RuyiNetLeaderboardVersionInfo>();
            foreach (var i in data.versions)
            {
                Versions.Add(new RuyiNetLeaderboardVersionInfo(i));
            }
        }

        /// <summary>
        /// The ID of the leaderboard.
        /// </summary>
        public string LeaderboardId { get; private set; }

        /// <summary>
        /// The type of leaderboard.
        /// </summary>
        public RuyiNetLeaderboardType LeaderboardType { get; private set; }

        /// <summary>
        /// The type of leaderboard rotation.
        /// </summary>
        public RuyiNetRotationType RotationType { get; private set; }

        /// <summary>
        /// The number of versions retained.
        /// </summary>
        public int RetainedCount { get; private set; }

        /// <summary>
        /// A list of versions currently retained.
        /// </summary>
        public List<RuyiNetLeaderboardVersionInfo> Versions { get; private set; }
    }
}
